Public/Invoke-DockerPull.ps1

```diff
@@ -36,5 +36,12 @@ function Invoke-DockerPull {
         }
         $imageToPull = "${postfixedRegistry}${ImageName}@${Digest}"
     }
-    Invoke-Command "docker pull ${imageToPull}"
+
+    $result = Invoke-Command "docker pull ${imageToPull}"
+    if ($result.Exitcode -ne 0) {
+        $message = "An error occured during docker pull. The error message was: ${result.Output}, the exit code was: ${result.Exitcode}"
+        Write-Debug "${message}"
+        throw "${message}"
+    }
+    return $result
```

Comment:
with return? can we decide on a style 😆

Reply:
As suggested, functions has consistent `return`s.